homeassistant/components/nederlandse_spoorwegen/sensor.py

```diff
@@ -128,40 +124,59 @@ def device_state_attributes(self):
             for k in self._trips[0].trip_parts:
                 route.append(k.destination)
 
-        return {
+        # Static attributes
+        attributes = {
             "going": self._trips[0].going,
             "departure_time_planned": self._trips[0].departure_time_planned.strftime(
                 "%H:%M"
             ),
-            "departure_time_actual": self._trips[0].departure_time_actual.strftime(
-                "%H:%M"
-            ),
-            "departure_delay": self._trips[0].departure_time_planned
-            != self._trips[0].departure_time_actual,
-            "departure_platform": self._trips[0].trip_parts[0].stops[0].platform,
-            "departure_platform_changed": self._trips[0]
-            .trip_parts[0]
-            .stops[0]
-            .platform_changed,
+            "departure_time_actual": None,
+            "departure_delay": False,
+            "departure_platform_planned": self._trips[0].departure_platform_planned,
+            "departure_platform_actual": None,
             "arrival_time_planned": self._trips[0].arrival_time_planned.strftime(
                 "%H:%M"
             ),
-            "arrival_time_actual": self._trips[0].arrival_time_actual.strftime("%H:%M"),
-            "arrival_delay": self._trips[0].arrival_time_planned
-            != self._trips[0].arrival_time_actual,
-            "arrival_platform": self._trips[0].trip_parts[0].stops[-1].platform,
-            "arrival_platform_changed": self._trips[0]
-            .trip_parts[0]
-            .stops[-1]
-            .platform_changed,
-            "next": self._trips[1].departure_time_actual.strftime("%H:%M"),
+            "arrival_time_actual": None,
+            "arrival_delay": False,
+            "arrival_platform_platform": self._trips[0].arrival_platform_planned,
+            "arrival_platform_actual": None,
+            "next": None,
             "status": self._trips[0].status.lower(),
             "transfers": self._trips[0].nr_transfers,
             "route": route,
-            "remarks": [r.message for r in self._trips[0].trip_remarks],
+            "remarks": None,
             ATTR_ATTRIBUTION: ATTRIBUTION,
         }
 
+        # Departure attributes
+        if self._trips[0].departure_time_actual is not None:
+            attributes["departure_time_actual"] = self._trips[
+                0
+            ].departure_time_actual.strftime("%H:%M")
+            attributes["departure_delay"] = True
```

Comment:
I wonder if this is right? The way I understand the API, "planned" and "actual" times can both be present even for normal operation and we only have a delay if "actual" is later than "planned".

Like here: https://github.com/aquatix/ns-api/blob/master/ns_api.py#L614

But I am not using this sensor so I will trust you if you say it isn't so. Just wanted to mention it.

Reply:
That's what I thought too at first, but (if I understand your point correctly) when there's no delay, `departure_time_actual` will be present but equal to None crashing the `strftime` function.

In fact, [https://github.com/aquatix/ns-api/blob/2d864ca3913a89996b631570a3d2a5a86e8b1164/ns_api.py#L554](https://github.com/aquatix/ns-api/blob/2d864ca3913a89996b631570a3d2a5a86e8b1164/ns_api.py#L554) is where you'll find that `departure_time_actual` can indeed be set to None.

This is the pattern I was trying to fix earlier, perhaps it could already be improved upon by changing nsapi (with a corresponding home-assistant PR). For the moment, let's get this sensor up and running again :)

Reply:
Yes, I think it is a good idea to approach upstream about this before making a lot of workarounds in Home Assistant.